alf/utils/spec_utils.py

```diff
@@ -92,7 +92,10 @@ def zeros_from_spec(nested_spec, batch_size):
         shape = [batch_size]
 
     def _zero_tensor(spec):
-        return spec.zeros(shape)
+        res = spec.zeros(shape)
+        if res.dtype == torch.float64:
```

Comment:
I think if torch.float64 is explicitly set as torch.float64, then we should honor the intention. This means that the right fix is to pass torch.float32 instead of torch.float64 when construct the spec.

Reply:
Agree. Now do the dtype conversion at the place when constructing the spec.